pom.xml

```diff
@@ -268,7 +286,94 @@
 					</execution>
 				</executions>
 			</plugin>
-        </plugins>
+			<plugin>
+				<groupId>org.jsonschema2pojo</groupId>
+				<artifactId>jsonschema2pojo-maven-plugin</artifactId>
+				<version>1.2.1</version>
+				<executions>
+					<execution>
+						<id>generate-artifact-packaged-from-schema</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>generate</goal>
+						</goals>
+						<!--<configuration>
+							<sourceDirectory>${project.basedir}/src/main/resources/schema</sourceDirectory>
+							<outputDirectory>${basedir}/src/main/java</outputDirectory>
+							<targetPackage>dev.cdevents.models</targetPackage>
+							<includeHashcodeAndEquals>false</includeHashcodeAndEquals>
+							<includeToString>false</includeToString>
+						</configuration>-->
+						<configuration>
+							<sourcePaths>
+								<sourcePath>${project.basedir}/src/main/resources/schema/artifact-packaged-event.json</sourcePath>
+							</sourcePaths>
+							<outputDirectory>${basedir}/src/main/java</outputDirectory>
+							<targetPackage>dev.cdevents.models.artifact.packaged</targetPackage>
+							<includeHashcodeAndEquals>false</includeHashcodeAndEquals>
+							<includeToString>false</includeToString>
+						</configuration>
+					</execution>
+					<execution>
+						<id>generate-artifact-published-from-schema</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>generate</goal>
+						</goals>
+						<configuration>
+							<sourcePaths>
+								<sourcePath>${project.basedir}/src/main/resources/schema/artifact-published-event.json</sourcePath>
+							</sourcePaths>
+							<outputDirectory>${basedir}/src/main/java</outputDirectory>
+							<targetPackage>dev.cdevents.models.artifact.published</targetPackage>
+							<includeHashcodeAndEquals>false</includeHashcodeAndEquals>
+							<includeToString>false</includeToString>
+						</configuration>
+					</execution>
+					<execution>
+						<id>generate-pipeline-run-finished-from-schema</id>
+						<phase>generate-sources</phase>
+						<goals>
+							<goal>generate</goal>
+						</goals>
+						<configuration>
+							<sourcePaths>
+								<sourcePath>${project.basedir}/src/main/resources/schema/pipeline-run-finished-event.json</sourcePath>
+							</sourcePaths>
+							<outputDirectory>${basedir}/src/main/java</outputDirectory>
+							<targetPackage>dev.cdevents.models.pipelinerun.finished</targetPackage>
+							<includeHashcodeAndEquals>false</includeHashcodeAndEquals>
+							<includeToString>false</includeToString>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<configuration>
+					<source>11</source>
+					<target>11</target>
+				</configuration>
+			</plugin>
+			<plugin>
+				<groupId>org.codehaus.mojo</groupId>

```

Comment:
Based on the previous comment on the mustache dependency. Might be worth separating code generation from generated code consumption. That is, migrate this project to a multi module Maven setup, putting code generators in one module, and cdevents Java sdk in another. Otherwise we risk exposing the code generator and its dependencies as part of the runtime dependencies required by consumers of cdevents-java-sdk.

Reply:
Updated the code as suggested with multi modules `generator` and `sdk` based on your [commit](https://github.com/aalmiray/sdk-java/commit/79e28295ce82579576ab69870d2e9da8e97e4a2d#diff-17d3fcce5ba64b158db9f92c868025524d9b4de3d8591f417b1dd59981ee51a7)